src/CodeFormatter.cs

```diff
@@ -347,5 +337,36 @@ private static void LogWorkspaceDiagnostics(ILogger logger, bool logWorkspaceWar
             options = optionsApplier.ApplyConventions(options, context.CurrentConventions, project.Language);
             return (document, options, context.CurrentConventions, true);
         }
+
+        private static async Task<bool> ShouldIgnoreDocument(
+            Document document, 
+            ImmutableHashSet<string> filesToFormat,
+            ImmutableHashSet<string> filesToIgnore,
+            CancellationToken cancellationToken)
+        {
+            if (!filesToFormat.IsEmpty && !filesToFormat.Contains(document.FilePath))
+            {
+               // If a files list was passed in, then ignore files not present in the list.
+                return true;
+            }
+            else if (!document.SupportsSyntaxTree)
+            {
+                return true;
+            }
+            else if (await GeneratedCodeUtilities.IsGeneratedCodeAsync(document, cancellationToken).ConfigureAwait(false))
+            {
+                // Ignore generated code files.
+                return true;
+            }
+            else if (!filesToIgnore.IsEmpty && filesToIgnore.Any(f => document.FilePath.Contains(f, StringComparison.OrdinalIgnoreCase)))
```

Comment:
Do we want to have ignore casing on this string compare? If so, should I also add it on :347

Reply:
Also, since the paths should be normalized, I think `.StartsWith` might be "more correct" here.

Reply:
Changed it to `StartsWith`, but I still need to keep the `StringComparison` bit since we're doing `document.FilePath.StartsWith`. Unlike the `ImmutableHashset`, `document.FilePath`'s comparer doesn't ignore casing.